huaweicloud/services/dataarts/resource_huaweicloud_dataarts_architecture_code_table_values.go

```diff
@@ -0,0 +1,369 @@
+				Required:    true,
+				ForceNew:    true,
+				Description: "The code of the code table field.",
+			},
+			"field_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The type of the code table field.",
+			},
+			"values": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"value": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "The value of a field.",
+						},
+						"ordinal": {
+							Type:        schema.TypeInt,
+							Computed:    true,
+							Description: "The ordinal of a value.",
+						},
+						"id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "The ID of a value.",
+						},
+					},
+				},
+				Description: "The values list of the code table field.",
+			},
+		},
+	}
+}
+
+func resourceArchitectureCodeTableValuesCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+
+	createCodeTableValuesProduct := "dataarts"
+	createCodeTableValuesClient, err := cfg.NewServiceClient(createCodeTableValuesProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Studio client: %s", err)
+	}
+
+	createCodeTableValuesPath, createCodeTableValuesOpt := buildCodeTableValuesPathAndOpt(createCodeTableValuesClient, d)
+	createCodeTableValuesOpt.JSONBody = utils.RemoveNil(buildCreateCodeTableValuesParams(d))
+	createCodeTableValuesResp, err := createCodeTableValuesClient.Request("PUT", createCodeTableValuesPath, &createCodeTableValuesOpt)
+	if err != nil {
+		return diag.Errorf("error creating DataArts Architecture code table values: %s", err)
+	}
+
+	_, err = utils.FlattenResponse(createCodeTableValuesResp)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	d.SetId(d.Get("field_name").(string))
+
+	return resourceArchitectureCodeTableValuesRead(ctx, d, meta)
+}
+
+func buildCreateCodeTableValuesParams(d *schema.ResourceData) map[string]interface{} {
+	params := make([]map[string]interface{}, 1)
+	params[0] = map[string]interface{}{
+		"id":                      d.Get("field_id"),
+		"ordinal":                 d.Get("field_ordinal"),
+		"name_ch":                 d.Get("field_name"),
+		"name_en":                 d.Get("field_code"),
+		"data_type":               d.Get("field_type"),
+		"code_table_field_values": buildCreateCodeTableFieldValuesParams(d),
+	}
+	return map[string]interface{}{
+		"to_add": params,
+	}
+}
+
+func buildCreateCodeTableFieldValuesParams(d *schema.ResourceData) []map[string]interface{} {
+	rawValues := d.Get("values").([]interface{})
+	values := make([]map[string]interface{}, len(rawValues))
+	for i, rawValue := range rawValues {
+		valueMap := rawValue.(map[string]interface{})
+		value := map[string]interface{}{
+			"fd_id":    d.Get("field_id"),
```

Comment:
we can get the value of `field_id` before the for loop

Reply:
fixed